src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -59,19 +62,50 @@ void check_sshfs_exists(mp::SSHSession& session)
     run_cmd(session, "which sshfs", error_handler);
 }
 
-void make_target_dir(mp::SSHSession& session, const std::string& target)
+// Split a path into existing and to-be-created parts.
+std::pair<std::string, std::string> get_path_split(mp::SSHSession& session, const std::string& target)
 {
-    run_cmd(session, fmt::format("sudo mkdir -p \"{}\"", target));
+    QDir complete_path(QString::fromStdString(target));
+    std::string absolute;
+
+    if (complete_path.isRelative())
+    {
+        std::string home = run_cmd(session, "pwd");
+        mp::utils::trim_end(home);
+        absolute = home + '/' + complete_path.path().toStdString();
+    }
+    else
+    {
+        absolute = complete_path.path().toStdString();
+    }
+
+    std::string existing =
+        run_cmd(session, fmt::format("/bin/bash -c 'P=\"{}\"; while [ ! -d \"$P/\" ]; do P=${{P%/*}}; done; echo $P/'",
+                                     absolute));
+    mp::utils::trim_end(existing);
+
+    return {existing,
+            QDir(QString::fromStdString(existing)).relativeFilePath(QString::fromStdString(absolute)).toStdString()};
 }
 
-void set_owner_for(mp::SSHSession& session, const std::string& target)
+// Create a directory on a given root folder.
+void make_target_dir(mp::SSHSession& session, const std::string& root, const std::string& relative_target)
+{
+    if (!relative_target.empty())
+        run_cmd(session, fmt::format("cd \"{}\" && sudo mkdir -p \"{}\"", root, relative_target));
```

Comment:
Probably best to wrap this in `sudo bash -c`, too.

Reply:
ok